src/pages/Integrations/Create/CardSelect.tsx

```diff
@@ -0,0 +1,175 @@
+  isReadOnly: boolean;
+}
+
+const CardSelect: React.FunctionComponent<CardSelectProps> = (
+  originalProps
+) => {
+  const {
+    isRequired,
+    label,
+    helperText,
+    hideLabel,
+    meta,
+    input,
+    options,
+    mutator,
+    DefaultIcon,
+    iconMapper,
+    ...props
+  } = useFieldApi(originalProps) as CardSelectProps;
+  const formOptions = useFormApi();
+  const [icons] = useState(() => {
+    const components = {};
+
+    options.forEach(
+      ({ value }) => (components[value] = iconMapper(value, DefaultIcon))
+    );
+
+    return components;
+  });
+
+  const isMulti = props.isMulti || props.multi;
+  const isDisabled = props.isDisabled || props.isReadOnly;
+  const inputValue = input.value || [];
+
+  const handleMulti = (value) =>
+    input.onChange(
+      inputValue.includes(value)
+        ? inputValue.filter((valueSelect) => valueSelect !== value)
+        : [...inputValue, value]
+    );
+
+  const handleSingle = (value) =>
+    input.onChange(inputValue === value ? undefined : value);
+
+  const handleClick = (value) =>
+    isMulti ? handleMulti(value) : handleSingle(value);
+
+  const onClick = (value) => {
+    if (isDisabled) {
+      return undefined;
+    }
+
+    handleClick(value);
+    input.onBlur();
+  };
+
+  const prepareCards = () =>
+    options
+      .map((option) => mutator(option, formOptions))
+      .map(({ value, label, isDisabled: itemIsDisabled }) => {
+        const disabled = itemIsDisabled || isDisabled;
+
+        if (!value) {
+          return undefined;
+        }
+
+        const Component = icons[value];
+
+        return (
+          <GridItem sm={6} md={4} key={value}>
+            <Tile
+              className={`src-c-wizard__tile${disabled ? ' disabled' : ''}`}
+              onClick={() => onClick(value)}
+              tabIndex={disabled ? -1 : 0}
+              onKeyPress={(e) => handleKeyPress(e, value, onClick)}
+              isDisabled={disabled}
+              title={label}
+              isStacked
+              isSelected={inputValue.includes(value)}
+              {...(Component && { icon: <Component /> })}
+            />
+          </GridItem>
+        );
+      });
+
+  const { error, touched } = meta;
+  const showError = touched && error;
+
+  return (
+    <FormGroup
+      isRequired={isRequired}
+      label={!hideLabel && label}
+      fieldId={input.name}
+      helperText={helperText}
+      helperTextInvalid={error}
+      validated={showError ? 'error' : 'default'}
+    >
+      <Grid hasGutter className="pf-u-mb-md">
+        {prepareCards()}
+      </Grid>
+      <FormHelperText>
+        <HelperText>
+          <HelperTextItem variant={showError ? 'error' : 'default'}>
+            {showError ? error : helperText}
+          </HelperTextItem>
+        </HelperText>
+      </FormHelperText>
+    </FormGroup>
+  );
+};
+
+CardSelect.defaultProps = {
+  options: [],
+  iconMapper: (_value, DefaultIcon) => DefaultIcon,
+  mutator: (x) => x,
+};
```

Comment:
Since this is a typescript we should utilize default values in function description.